packages/react-composites/src/composites/CallComposite/utils/Utils.ts

```diff
@@ -542,3 +548,27 @@ export const getBackgroundEffectFromSelectedEffect = (
  */
 export const getSelectedCameraFromAdapterState = (state: CallAdapterState): VideoDeviceInfo | undefined =>
   state.devices.selectedCamera || state.devices.cameras[0];
+
+/**
+ * helper to determine if the adapter has a locator or targetCallees
+ * @param locatorOrTargetCallees
+ * @returns enum to determine if the adapter has a locator or targetCallees
+ * @private
+ */
+export const getLocatorOrTargetCallees = (
```

Comment:
Use `is` type narrowing to return a type scoped object instead of string that needs to be subsequently used to type cast.
We do this in some other places for reference:
![image](https://github.com/Azure/communication-ui-library/assets/2684369/9dcfbb88-7d10-4af3-8ae1-ecd027633eca)

So this would become something like:
```ts
const isTargetCalleeList = (locatorOrTargetCallees: CallAdapterLocator
    | (
        | MicrosoftTeamsAppIdentifier
        | /* @conditional-compile-remove(PSTN-calls) */ PhoneNumberIdentifier
        | /* @conditional-compile-remove(one-to-n-calling) */ CommunicationUserIdentifier
        | /* @conditional-compile-remove(teams-adhoc-call) */ MicrosoftTeamsUserIdentifier
        | UnknownIdentifier
      )[]) => Array.isArray(locatorOrTargetCallees);


// Then when used:
if (isTargetCalleeList(obj)) {
  // obj inside this scope is now auto typecasted correctly as a targetCallee list
} else {
  // obj is now auto typecasted as the locator type
}
```

Reply:
(note that's totally untested, but typescript should support the auto type narrowing)

Reply:
This works and is super cool! Change has been made!